go/azure_identity_credential_adapter.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidext
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/go-autorest/autorest"
+)
+
+type tokenAdapter struct {
+	Token string `json:"access_token"`
+}
+
+// NewAzureIdentityCredentialAdapter is used to adapt an azcore.Credential to an autorest.Authorizer
+func NewAzureIdentityCredentialAdapter(credential azcore.TokenCredential, options azcore.TokenRequestOptions) (autorest.Authorizer, error) {
```

Comment:
Shouldn't the user pass in the context as a param here? @jhendrixMSFT this is performing an I/O operation, I think we should make it clear, what do you think?

Reply:
You are correct, however we can't ship with this current implementation anyways (see my other comment about using the authentication policy).